src/Resistor/ResistorInput.tsx

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import { ResistorColorEntry as Color, Resistor } from 'puzzle-lib';
+import { DropdownButton, MenuItem, ButtonGroup } from 'react-bootstrap';
+import './ResistorInput.css';
+
+type Props = {
+};
+
+type State = {
+  value: String,
+  bands: (Color | null)[]
+};
+
+class ResistorInput extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      value: '0',
+      bands: [Resistor.BLACK, Resistor.BLACK, Resistor.BLACK, null, Resistor.BROWN]
+    };
+  }
+
+  public render() {
+    return (
+      <div className="ResistorInput">
+        <div className="ResistorDisplay">
+          <svg width="240" viewBox="0 0 136 40">
+            <defs>
+              <g id="Wire">
+                <rect y="20" height="1" width="8" fill="black" />
+              </g>
+              <g id="Resistor">
+                <rect
+                  y="8"
+                  x="8"
+                  width="120"
+                  height="24"
+                  rx="4"
+                  ry="4"
+                  stroke="black"
+                  strokeWidth="0.5"
+                  fill="burlywood"
+                />
+                <use xlinkHref="#Wire" x="0" />
+                <use xlinkHref="#Wire" x="128" />
+              </g>
+              <g id="Band">
+                <rect y="8" height="24" width="12" stroke="black" strokeWidth="0.5" />
+              </g>
+            </defs>
+            <use xlinkHref="#Resistor" x="0" y="0" />
+            {this.getBand(0)}
+            {this.getBand(1)}
+            {this.getBand(2)}
+            {this.getBand(3)}
+            {this.getBand(4)}
+            Sorry, your browser does not support inline SVG.
+          </svg>
+        </div>
+        <div className="ResistorValue">Value: {this.state.value}</div>
+        <div className="ResistorSelectors">
+          <ButtonGroup>
+            {this.valueSelector(0, 'First Band')}
+            {this.valueSelector(1, 'Second Band')}
+            {this.multiplierSelector(2, 'Third Band')}
+            {this.multiplierOrNullSelector(3, 'Fourth Band')}
+            {this.toleranceSelector(4, 'Tolerance Band')}
+          </ButtonGroup>
+        </div>
+      </div>
+    );
+  }
+
+  private getBand(index: number) {
+    const band = this.state.bands[index];
+    if (band != null) {
+      return (
+        <use 
+          xlinkHref="#Band"
+          x={this.getBandOffset(index)}
+          fill={band!.colorCode}

```

Comment:
Is the `!` still required?

Reply:
Yes, since band could still be null.

Reply:
...how? Isn't that what is being checked right above?

Reply:
Hence my confusion with the linter.  !== seems to have gotten it to parse correctly.